File: WorldFeed/Server/src/Services/Identity/WorldFeed.Identity.API/DTO/UserIdentifierDTO.cs
namespace WorldFeed.Identity.API.DTO
{
    using Newtonsoft.Json;

    using WorldFeed.Common.Public.Models.Interfaces;

    public class UserIdentifierDTO : IUserIdentifier
    {
        private long _id;

        [JsonProperty("id")]
        public long Id
        {
            get => _id;
            set
            {
                _id = value;
                IdStr = _id.ToString();
            }
        }

        [JsonProperty("id_str")]
        public string IdStr { get; set; }

        [JsonProperty("screen_name")]
        public string ScreenName { get; set; }
    }
}

File: WorldFeed/Client/src/app/core/Core/DTO/UserIdentifierDTO.ts
import {IUserIdentifier} from "../../Public/Models/Interfaces/IUserIdentifier";

export class UserIdentifierDTO implements IUserIdentifier {
  private _id: number;

  // [JsonProperty("id")]
  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
    this.idStr = this._id.toString();
  }

  // [JsonProperty("id_str")]
  public idStr: string;

  // [JsonProperty("screen_name")]
  public screenName: string;
}
